src/movie/movie.service.ts

```diff
@@ -12,15 +12,8 @@ export class MovieService {
 
   async getMovieByImdbId(imdbId: string): Promise<Movie> {
     try {
-      // get metadata
-      const metadataUrl = `${this.METADATA_SERVICE}?imdbId=${imdbId}`
-      const metadataResponse = await axios.get(metadataUrl)
-      const metadata = Utils.extractDataFromMetadataResponse(metadataResponse)
-
-      // get poster url
-      const posterUrl = `${this.POSTER_SERVICE}?id=${imdbId}`
-      const posterResponse = await axios.get(posterUrl)
-      const poster = Utils.extractDataFromPosterResponse(posterResponse)
+      const metadata = (await this.getMetadata(imdbId))[0]
```

Comment:
While it is still not nice, this would be caught by the try/catch

Reply:
So what do you propose? Adding a separate try catch and throw a specific error message?